File: library/src/main/java/rapid/decoder/frame/AspectRatioCalculator.java
package rapid.decoder.frame;

public class AspectRatioCalculator {
	@SuppressWarnings("UnusedDeclaration")
    public static int getHeight(int width, int height, int targetWidth) {
		final double ratio = (double) height / width;
		return (int) Math.round(ratio * targetWidth);
	}
	
	@SuppressWarnings("UnusedDeclaration")
    public static int getWidth(int width, int height, int targetHeight) {
		final double ratio = (double) width / height;
		return (int) Math.round(ratio * targetHeight);
	}
	
	public static float getHeight(float width, float height, float targetWidth) {
		return (height / width) * targetWidth;
	}
	
	public static float getWidth(float width, float height, float targetHeight) {
		return (width / height) * targetHeight;
	}
}

File: decoder/src/main/java/rapiddecoder/util/AspectRatioCalculator.kt
package rapiddecoder.util

object AspectRatioCalculator {
    fun getHeight(width: Int, height: Int, targetWidth: Int): Int {
        val ratio = height.toDouble() / width
        return Math.round(ratio * targetWidth).toInt()
    }

    fun getWidth(width: Int, height: Int, targetHeight: Int): Int {
        val ratio = width.toDouble() / height
        return Math.round(ratio * targetHeight).toInt()
    }

    fun getHeight(width: Float, height: Float, targetWidth: Float): Float =
            height / width * targetWidth

    fun getWidth(width: Float, height: Float, targetHeight: Float): Float =
            width / height * targetHeight
}
